pkg/agent/handler/accessrequest.go

```diff
@@ -56,6 +58,11 @@ func (h *accessRequestHandler) Handle(ctx context.Context, meta *proto.EventMeta
 		return nil
 	}
 
+	// add or update the cache with the access request
+	if action == proto.Event_CREATED || action == proto.Event_UPDATED {
```

Comment:
How come the resource gets cached before processing?

Reply:
Want to make sure our cache has the latests, on an update event we may not have to process anything but want the cache updated